### PR TITLE
Update DBT-2 playbooks

### PR DESCRIPTION
**Pull request by mw2q**
Several updates for ease of benchmarking and to prepare the system for more in-depth system characterization.

* Use database operating system user on driver and client systems instead of separately created dbt2 user
   * Create passwordless ssh keys on driver systems
   * Set up authorized_keys on client and database system with driver users' ssh keys
   * Set up known_hosts file on driver systems with client systems and database systems id
* Trust connections from driver systems to the database (for 2-tier configurations)
* Install additional packages (perf, rsync, tmux)

### CODE REVIEW

File: roles/setup_dbt2_driver/tasks/main.yml [hunk 1]
```diff
@@ -8,11 +8,35 @@
...
+    ssh_targets: "{{ ssh_targets + [ hostvars[item]['private_ip'] ] }}"
+  with_items: "{{ groups['dbt2_client'] }}"
+  when: "{{ 'dbt2_client' in groups }}"
+  register: ssh_targets_list
+
+- name: Make a list of database system addresses
+  set_fact:
+    ssh_targets: "{{ ssh_targets + [ hostvars[item]['private_ip'] ] }}"
+  with_items: "{{ groups['primary'] }}"
+  register: ssh_targets_list
```
Comment (jt-edb) [inline roles/setup_dbt2_driver/tasks/main.yml:29]: Not sure you have to use `register`, you can use directly `ssh_targets`:
```yaml
- name: Make a list of DBT-2 client system addresses
  set_fact:
    ssh_targets: "{{ ssh_targets|default([]) + [ hostvars[item]['private_ip'] ] }}"
  with_items: "{{ groups['dbt2_client'] }}"
  when: "{{ 'dbt2_client' in groups }}"

- name: Make a list of database system addresses
  set_fact:
    ssh_targets: "{{ ssh_targets|default([]) + [ hostvars[item]['private_ip'] ] }}"
  with_items: "{{ groups['primary'] }}"

...

- name: Gather SSH public keys from DBT-2 client and database systems.
  ansible.builtin.command: ssh-keyscan {{ item }}
  register: ssh_keyscan
  loop: "{{ ssh_targets }}"
  when: "{{ 'dbt2_client' in groups }}"
```

File: roles/setup_dbt2/tasks/dbt2_configure_pgpass.yml [hunk 2]
```diff
@@ -1,24 +1,10 @@
...
-    dest: "{{ dbt2_user_home }}/.pgpass"
+    dest: "{{ pg_user_home }}/.pgpass"
     mode: "0600"
     src: pgpass.template
   vars:
-    pg_superuser_override: "{{ pg_dbt2_user }}"
-    pg_superuser_password_override: "{{ pg_dbt2_user_password }}"
-  become_user: "{{ dbt2_user }}"
+    pg_superuser_override: "{{ pg_owner }}"
+    pg_superuser_password_override: "{{ pg_superuser_password }}"
```
[the remark this answers is not in the document]
Comment (mw2q) [reply roles/setup_dbt2/tasks/dbt2_configure_pgpass.yml:9]: Ok.

File: roles/setup_dbt2_client/defaults/main.yml [hunk 3]
```diff
@@ -1,8 +1,2 @@
 ---
-dbt2_user: dbt2
-dbt2_user_home: /home/dbt2
 dbt2_client_port: 30000
-pg_dbt2_user: dbt2
-pg_dbt2_dbname: dbt2
-pg_dbt2_user_password: ""
-pg_version: 14
```
[the remark this answers is not in the document]
Comment (mw2q) [reply roles/setup_dbt2_client/defaults/main.yml:8]: I'm not sure if I'm fully understanding.  I removed the variables that I wanted to be inherited from other roles.  It seems like if I set these, specifically the ones set to empty string, then what I inherit doesn't get inherited anymore.  So I think I'm still not doing something right...

File: roles/setup_dbt2_driver/tasks/main.yml (hunk 1 above)
Comment (mw2q) [reply roles/setup_dbt2_driver/tasks/main.yml]: Ok, cleaned that up.

File: roles/setup_dbt2_client/defaults/main.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jt-edb) [reply roles/setup_dbt2_client/defaults/main.yml:8]: Actually, we don't want the variables to be inherited from other roles: each role can be used independently from the others. For example, to set up **only** a DBT2 client server we should be able to use the following playbook:
```yaml
---
- hosts: dbt2_client
  name: DBT2 client setup playbook
  become: yes
  gather_facts: yes

  collections:
    - edb_devops.edb_postgres

  pre_tasks:
    - name: Initialize the user defined variables
      set_fact:
        pg_user_home: /var/lib/pgsql
        pg_version: 14
        pg_owner: postgres

  roles:
    - setup_dbt2_client
```

Does this make sense?

**Comment by mw2q** [comment]
I think I've set all of the missing variables.  One thing that I had forgotten was that some of them were being set in the vars/ files, as opposed to defaults/ files.  I think I was mistaken that they were being inherited from other roles, and I was in fact setting some of them from with the local roles.

**Comment by mw2q** [comment]
@jt-edb fyi, rebased against main c2c589b23742873c6281eb83482963d89afa27f6

**Comment by mw2q** [comment]
@jt-edb Took another stab and making sure variables are defined in the defaults/main.yml

Also updated README files.

File: roles/setup_dbt2/README.md [hunk 4]
```diff
@@ -1,13 +1,103 @@
...
+    - edb_devops.edb_postgres
+
+  roles:
+    - role: setup_repo
+      when: "'setup_repo' in lookup('edb_devops.edb_postgres.supported_roles', wantlist=True)"
+    - role: install_dbserver
+      when: "'install_dbserver' in lookup('edb_devops.edb_postgres.supported_roles', wantlist=True)"
+    - role: init_dbserver
+      when: "'init_dbserver' in lookup('edb_devops.edb_postgres.supported_roles', wantlist=True)"
+      when: "'setup_dbt2' in lookup('edb_devops.edb_postgres.supported_roles', wantlist=True)"
```
Comment (jt-edb) [inline roles/setup_dbt2/README.md:86]: The `- role: setup_dbt2` line is missing here.

**Comment by jt-edb** [comment]
Hi @mw2q 

Please find below the summary of the failing tests and warnings.

```
setup_dbt2_client-ansible-tester-1  | TASK [edb_devops.edb_postgres.setup_dbt2_driver : Make a list of DBT-2 client system addresses] ***
setup_dbt2_client-ansible-tester-1  | skipping: [primary1] => (item=dbt2_client1) 
Warning: 2_client-ansible-tester-1  | [WARNING]: conditional statements should not include jinja2 templating
setup_dbt2_client-ansible-tester-1  | delimiters such as {{ }} or {% %}. Found: {{ 'dbt2_client' in groups }}
```
```
setup_dbt2_client-ansible-tester-1  | TASK [edb_devops.edb_postgres.setup_dbt2_driver : Make a list of database system addresses] ***
setup_dbt2_client-ansible-tester-1  | skipping: [primary1] => (item=primary1) 
Warning: 2_client-ansible-tester-1  | [WARNING]: conditional statements should not include jinja2 templating
setup_dbt2_client-ansible-tester-1  | delimiters such as {{ }} or {% %}. Found: {{ 'primary' in groups }}
```
```
setup_dbt2-ansible-tester-1  | TASK [edb_devops.edb_postgres.setup_dbt2 : Gather DBT-2 client addresses] ******
Warning: 2-ansible-tester-1  | [WARNING]: conditional statements should not include jinja2 templating
setup_dbt2-ansible-tester-1  | delimiters such as {{ }} or {% %}. Found: {{ 'dbt2_client' in groups }}
setup_dbt2-ansible-tester-1  | ok: [primary1] => (item=dbt2_client1)
```
```
setup_dbt2-ansible-tester-1  | TASK [edb_devops.edb_postgres.setup_dbt2 : Make a list of DBT-2 client addresses] ***
Warning: 2-ansible-tester-1  | [WARNING]: conditional statements should not include jinja2 templating
setup_dbt2-ansible-tester-1  | delimiters such as {{ }} or {% %}. Found: {{ 'dbt2_client' in groups }}
setup_dbt2-ansible-tester-1  | ok: [primary1]
```
```
setup_dbt2-ansible-tester-1  | TASK [Update pg_hba.conf for access from the DBT-2 clients' private IP addresses] ***
Warning: 2-ansible-tester-1  | [WARNING]: conditional statements should not include jinja2 templating
setup_dbt2-ansible-tester-1  | delimiters such as {{ }} or {% %}. Found: {{ 'dbt2_client' in groups }}
```
```
setup_dbt2_client-ansible-tester-1  | TASK [edb_devops.edb_postgres.setup_dbt2 : Ensure postgres's SSH public key from all drivers are installed] ***
setup_dbt2_client-ansible-tester-1  | fatal: [dbt2_client1]: FAILED! => {"changed": false, "gid": 1000, "group": "postgres", "mode": "0644", "msg": "chgrp failed: failed to look up group ", "owner": "postgres", "path": "/var/lib/pgsql/.ssh/authorized_keys", "size": 587, "state": "file", "uid": 1000}
```

File: roles/setup_dbt2/README.md (hunk 4 above)
Comment (mw2q) [reply roles/setup_dbt2/README.md:86]: Fixed.

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**